arwen/contexts/runtime.go

```diff
@@ -188,7 +189,43 @@ func (context *runtimeContext) GetVMInput() *vmcommon.VMInput {
 }
 
 func (context *runtimeContext) SetVMInput(vmInput *vmcommon.VMInput) {
-	context.vmInput = vmInput
+	if !context.host.IsArwenV2Enabled() || vmInput == nil {
+		context.vmInput = vmInput
+		return
+	}
+
+	context.vmInput = &vmcommon.VMInput{
+		CallType:    vmInput.CallType,
+		GasPrice:    vmInput.GasPrice,
+		GasProvided: vmInput.GasProvided,
+	}
+
+	if vmInput.CallValue != nil {
+		context.vmInput.CallValue = big.NewInt(0).Set(vmInput.CallValue)
+	}
+
+	if len(vmInput.CallerAddr) > 0 {
+		context.vmInput.CallerAddr = make([]byte, len(vmInput.CallerAddr))
+		copy(context.vmInput.CallerAddr, vmInput.CallerAddr)
+	}
+
+	if len(vmInput.OriginalTxHash) > 0 {
+		context.vmInput.OriginalTxHash = make([]byte, len(vmInput.OriginalTxHash))
+		copy(context.vmInput.OriginalTxHash, vmInput.OriginalTxHash)
+	}
+
+	if len(vmInput.CurrentTxHash) > 0 {
+		context.vmInput.CurrentTxHash = make([]byte, len(vmInput.CurrentTxHash))
+		copy(context.vmInput.CurrentTxHash, vmInput.CurrentTxHash)
+	}
+
+	if len(vmInput.Arguments) > 0 {
```

Comment:
Perhaps copying just the pointer would have also been safe (since arguments are immutable) - correct?

Reply:
I better copy as they were come errors, unforseen rewrites